v3/src/components/codap-modal.tsx

```diff
@@ -15,28 +15,122 @@ interface IProps {
 }
 
 export const CodapModal = forwardRef(({
-  children, closeOnOverlayClick=true, initialRef, isOpen, onClick, onClose, modalWidth, modalHeight
-}: IProps, ref: React.LegacyRef<HTMLElement> | undefined) => {
+  children, initialRef, isOpen, onClick, onClose, modalWidth, modalHeight
+}: IProps, ref: React.Ref<HTMLElement> | undefined) => {
+
   return (
     <Modal
-      closeOnOverlayClick={closeOnOverlayClick}
       data-testid="codap-modal"
       initialFocusRef={initialRef}
       isOpen={isOpen}
       onClose={onClose}
       size="xs"
     >
       <ModalOverlay />
-      <ModalContent
-        className="codap-modal-content"
-        h={modalHeight || "500px"}
-        onClick={onClick}
-        ref={ref}
-        w={modalWidth || "400px"}
-      >
-        {children}
-      </ModalContent>
+      <DraggableModalContent
+          fRef={ref}
+          modalWidth={modalWidth}
+          modalHeight={modalHeight}
+          onClick={onClick}
+          isOpen={isOpen}
+        >
+          {children}
+      </DraggableModalContent>
     </Modal>
   )
 })
 CodapModal.displayName = "CodapModal"
+
+interface IDraggableModalContentProps {
+  children: ReactNode
+  modalWidth?: string
+  modalHeight?: string
+  onClick?: () => void
+  fRef: React.Ref<HTMLElement> | undefined
+  isOpen: boolean
+}
+
+const DraggableModalContent = ({children, modalWidth, modalHeight, onClick, fRef, isOpen
+    }: IDraggableModalContentProps) => {
+  const [modalPos, setModalPos] = useState({left: 350, top: 250})
+  const modalRef = useRef<HTMLElement | null>(null)
+  const mergedRef = useMergeRefs(fRef, modalRef)
+
+  useEffect(() => {
+    const modalElement = modalRef.current
+    if (!modalElement) {
+      return
+    }
+
+    let isDragging = false
+    let startX = 0
+    let startY = 0
+    let initialX = 0
+    let initialY = 0
+
+    const onMouseDown = (e: MouseEvent) => {
```

Comment:
One thing I forgot to mention on my earlier review -- the general convention for event handler functions like this is that the implementing functions are named `handleEvent` and the props used to pass them to other components are named `onEvent`. So this function and its brethren would more properly be:
```suggestion
    const handleMouseDown = (e: MouseEvent) => {
```
and then down below when passed as a prop it would be `onMouseDown={handleMouseDown}`.